CMakeLists.txt

```diff
@@ -22,9 +24,19 @@ if (PROJECT_IS_TOP_LEVEL)
   set(ATEN_CUDA_ROOT "${TORCH_INSTALL_PREFIX}/include/ATen")
   # CXX flags is necessary since https://github.com/pytorch/pytorch/issues/98093
   string(APPEND CMAKE_CXX_FLAGS ${TORCH_CXX_FLAGS})
+  if(BUILD_TEST)
+    if(SYSTEM_GTEST)
+      find_package(GTest REQUIRED)
+    else()
+      add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/googletest)
+    endif()
+  endif()
   if(BUILD_NVFUSER_BENCHMARK)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/googletest)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/benchmark)
+    if(SYSTEM_GBENCH)
```

Comment:
Partial support for this exists here: https://github.com/NVIDIA/Fuser/blob/main/Dependencies.cmake#L33

Reply:
WOW, thanks for doing that!